### PR TITLE
Set `force_latest_compat=true` by default on Dependabot/CompatHelper PRs, and set `force_latest_compat=false` by default otherwise

### PR DESCRIPTION
**Pull request by DilumAluthge**
This pull request is made against #2176 

You can view the correct diff (#2234 against #2176) at this link: https://github.com/DilumAluthge/Pkg.jl/compare/dpa/force-latest-compat...dpa/autodetect-dependabot

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@christopher-dG Can you review this?

**Comment by christopher-dG** [comment]
I think you set the target branch to master.

**Comment by DilumAluthge** [comment]
> I think you set the target branch to master.

Yeah. GitHub won't let me set the target (base) branch to `dpa/force-latest-compat` because the branch is in a fork.

**Comment by DilumAluthge** [comment]
You can view the correct diff (#2234 against #2176) at this link: https://github.com/DilumAluthge/Pkg.jl/compare/dpa/force-latest-compat...dpa/autodetect-dependabot

File: src/Operations.jl [hunk 1]
```diff
@@ -1390,7 +1391,7 @@ end
 
 # ctx + pkg used to compute parent dep graph
 function sandbox(fn::Function, ctx::Context, target::PackageSpec, target_path::String,
-                 sandbox_path::String, sandbox_project_override)
+                 sandbox_path::String, sandbox_project_override; force_latest_compat::Union{Bool, Symbol}=:autodetect)
```
Comment (christopher-dG) [inline src/Operations.jl:1394]: Maybe it would be cleaner to just do `::Bool=Types.decide_force_latest_compat()`?

Comment (DilumAluthge) [reply src/Operations.jl:1394]: Do you mean `::Bool=Types.is_dependabot_job()`

Comment (christopher-dG) [reply src/Operations.jl:1394]: Oh yeah, sorry!

**Comment by KristofferC** [comment]
I must say, I am quite uncomfortable coding stuff like this into Pkg. There must be some better option than this. Like having this logic in the CI file and pass the argument based on that.

**Comment by fredrikekre** [comment]
I agree (had a similar comment already written in a review that I failed to post...).

**Comment by christopher-dG** [comment]
The problem with that is it's 1 fix vs N fixes where N is the number of Julia packages. As great as it would be if everyone instantly updated their CI configs when it was needed, that just doesn't happen. Which is fine, since it's their responsibility, but it's still going to annoy and confuse loads of people.

**Comment by KristofferC** [comment]
CompatBot could try update the CI file?

**Comment by DilumAluthge** [comment]
> CompatBot could try update the CI file?

It would have to be Dependabot. @christopher-dG Is that feasible?

**Comment by DilumAluthge** [comment]
Actually, here's one option: if users use the `julia-runtest` action, we could implement this "autodetect" logic there.

This would automatically cover most people using GitHub Actions.

Travis and AppVeyor people will need to fix their scripts manually.

**Comment by christopher-dG** [comment]
I think encoding this in CI scripts (the ones that go in people's repos) is a non-starter. Doing it once per CI provider seems okay, and it's not tied to Julia releases then.

**Comment by DilumAluthge** [comment]
@christopher-dG If it's okay with you, I'll close this PR, and I'll work on adding this "autodetect" functionality into the `julia-runtest` GitHub Action.

**Comment by DilumAluthge** [comment]
I've moved the "auto-detect Dependabot/CompatHelper" functionality into https://github.com/julia-actions/julia-runtest/pull/20, so I have closed this PR.

**Comment by SaschaMann** [comment]
> Travis and AppVeyor people will need to fix their scripts manually.

You could potentially add it to Travis here: https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/julia.rb

**Comment by christopher-dG** [comment]
Pretty much everyone using AppVeyor is doing so via [AppVeyor.jl](https://github.com/JuliaCI/Appveyor.jl). GHA+Travis+AppVeyor probably covers >95% of open source package CI.

**Comment by DilumAluthge** [comment]
See https://github.com/JuliaRegistries/CompatHelper.jl/issues/298 to track the deployment of the `force_latest_compatible_version` feature across the most commonly used CI platforms.

**Pull request closed without merging**